flink-libraries/flink-table/src/main/scala/org/apache/flink/api/table/plan/rules/logical/FlinkJoinUnionTransposeRule.scala

```diff
@@ -0,0 +1,103 @@
+/* 
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.api.table.plan.rules.logical
+
+import org.apache.calcite.plan.RelOptRule.{any, operand, convert => convertTrait}
+import org.apache.calcite.plan.RelOptRule
+import org.apache.calcite.plan.RelOptRuleOperand
+import org.apache.calcite.plan.RelOptRuleCall
+import org.apache.calcite.rel.RelNode
+import java.util.ArrayList
+import scala.collection.JavaConversions._
+import org.apache.calcite.rel.logical.LogicalJoin
+import org.apache.calcite.rel.logical.LogicalUnion
+import org.apache.calcite.rel.core.Join
+import org.apache.calcite.rel.core.Union
+
+class FlinkJoinUnionTransposeRule(
```

Comment:
Can you add a notice that this file was copied from Apache Calcite and briefly describe what you changed and why?